src/lib/ArrayInfo.ts

```diff
@@ -1,121 +1,179 @@
-import {DataError} from "./Errors"
-import {KeyValue} from "./Utils"
+import { DataError } from './Errors'
+import { KeyValue } from './Utils'
 
 function isInt(value: any) {
-    return !isNaN(value) &&
-        Number(value) == value && !isNaN(parseInt(value, 10))
+  return !isNaN(value) && Number(value) == value && !isNaN(parseInt(value, 10))
 }
 
-export const arrayRegex = () => /^([\.0-9a-zA-Z_$\-][0-9a-zA-Z_\-$\.]*)\[((?!(\]|\[)).*|)\]$/gm
+export const arrayRegex = () => /^([\.0-9a-zA-Z_$\-][0-9a-zA-Z_\-$\.]*)(.*)/gm
+export const arrayIndiciesRegex = /\[(.*?)\]/g
```

Comment:
If we split the string that is match on `[` we can then force this regex to start with `[`.

Reply:
Additional function logic was added to figure out the nested array syntax and to support the 'append' logic.